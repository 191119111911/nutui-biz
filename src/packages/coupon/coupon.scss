@mixin ellipsis {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}
.nut-biz-coupon__box {
  display: flex;
  background-size: 100% 100%;
  position: relative;
  .nut-biz-coupon__main {
    position: relative;
    display: flex;
    padding: 16px;
    align-items: center;
    text-align: left;
    &-price {
      display: flex;
      align-items: baseline;
      margin-right: 20px;
      font-size: 32px;

      &--currency {
        font-size: 16px;
      }
    }
    &-content {
      overflow: hidden;
      &--maintitle {
        font-size: 14px;
        @include ellipsis;
      }
      &--subtitle,
      &--timerange {
        font-size: 12px;
        width: 100%;
        @include ellipsis;
      }
    }
    &-usedicon {
      position: absolute;
      top: 0px;
      right: 7px;
    }
  }
  .nut-biz-coupon__btns {
    flex: 1;
    display: flex;
    justify-content: center;
    align-items: center;
  }
  .nut-biz-coupon__label {
    position: absolute;
    top: 0px;
    left: 0px;
    height: 14px;
    line-height: 14px;
    padding: 0 2px;
    font-size: 10px;
    font-weight: 400;
    color: red;
    background: #fff;
    border-radius: 6px 0;
    max-width: 100px;
    @include ellipsis;
  }
}
.nut-biz-coupon__box--small {
  height: 81px;
  .nut-biz-coupon__main {
    color: red;
    flex-direction: column;
    text-align: center;
    padding: 13px 7px 10px;
    &-price {
      font-size: 18px;
      display: flex;
      justify-content: center;
      margin-right: 0;
      &--currency {
        font-size: 12px;
      }
      &--number {
        font-size: 18px;
      }
    }
    &-content {
      width: 100%;
      @include ellipsis;
      &--maintitle {
        font-size: 10px;
      }
      &--subtitle {
        color: rgba(250, 44, 25, 0.5);
        font-size: 10px;
        @include ellipsis;
      }
    }
  }
  .nut-biz-coupon__btns {
    &-vertcal {
      display: flex;
      align-items: center;
      height: 100%;
      font-size: 10px;
      padding: 0 7px;
      color: red;
    }
  }
  .nut-biz-coupon__label {
    background: red;
    color: #fff;
  }
}
