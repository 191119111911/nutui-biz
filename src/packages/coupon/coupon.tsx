import React, { FunctionComponent, useEffect, ReactNode } from "react";
import { IComponent } from "@/utils/typings";
import { Button, ButtonProps } from "@nutui/nutui-react";
import { throttle } from "@/utils/throttle";
import bem from "@/utils/bem";

export interface ICouponType {
  price: number; //优惠券价格或者折扣价
  currency: string; //货币符号
  mainTitle: string; //主标题
  subTitle: string; //副标题
  timeRange?: string; //优惠券使用时间范围
  label: string; //优惠券左上角的标签内容
}

export type CouponType = "large" | "small";
export type IPricePosition = "front" | "back";

export interface CouponProps extends IComponent {
  type: CouponType; //优惠券的尺寸
  couponMarginRight: number; //每张优惠券的margin右边距
  couponMarginBottom: number; //每张优惠券的margin下边距
  couponWidth: string; //每张优惠券的宽度 30px 或者 30%
  couponHeight: string; //每张优惠券的高度
  couponMainWidth: string; //优惠券左侧的宽度10px或者10%
  couponMainColor: string; //优惠券左侧的文字颜色
  couponBgImg: string; //优惠券的背景图
  pricePosition: IPricePosition; //价格和标签的前后位置
  coupon: ICouponType; //渲染优惠券内容
  btnText: string; //按钮文案
  isReceived: boolean; //是否领取
  receivedBtnText: string; //领取后的按钮文案
  className: string;
  buttonProps: Partial<ButtonProps>; //按钮props
  usedIcon: ReactNode;
  onClick: any;
  itemData: any; //父组件传递过来的数据，用户函数参数返回
}
const couponObj = {
  price: 12,
  currency: "¥",
  mainTitle: "主标题内容",
  subTitle: "用于渲染副标题文案",
  timeRange: "2022.03.01-2022.04.01",
  label: "内购专享",
} as ICouponType;
const defaultProps = {
  type: "large",
  couponMarginRight: 0,
  couponMarginBottom: 0,
  couponWidth: "100%",
  couponHeight: "auto",
  couponMainWidth: "69%",
  couponMainColor: "#fff",
  couponBgImg:
    "https://storage.360buyimg.com/jdcdkh/open/1.0.0/assets/bg-coupon-red.f6ae2e19.png",
  pricePosition: "back",
  coupon: couponObj,
  btnText: "立即领取",
  isReceived: false,
  receivedBtnText: "已经领取",
} as CouponProps;

export const Coupon: FunctionComponent<
  Partial<CouponProps> & Omit<React.HTMLAttributes<HTMLDivElement>, "onChange">
> = React.memo((props) => {
  const {
    type,
    couponWidth,
    couponHeight,
    couponMarginRight,
    couponMarginBottom,
    couponMainWidth,
    couponMainColor,
    couponBgImg,
    pricePosition,
    coupon,
    btnText,
    isReceived,
    receivedBtnText,
    className,
    usedIcon,
    onClick,
    itemData,
    buttonProps,
    ...rest
  } = {
    ...defaultProps,
    ...props,
  };
  const b = bem("biz-coupon");
  const disabdb = bem("biz-dis-coupon");
  useEffect(() => {}, []);
  const renderPrice = () => {
    if (pricePosition === "front") {
      return (
        <div className={b("main-price")}>
          <div className={b("main-price--number")}>
            {coupon?.price.toString().substring(0, 5)}
          </div>
          <div className={b("main-price--currency")}>{coupon?.currency}</div>
        </div>
      );
    } else {
      return (
        <div className={b("main-price")}>
          <div className={b("main-price--currency")}>{coupon?.currency}</div>
          <div className={b("main-price--number")}>
            {coupon?.price.toString().substring(0, 5)}
          </div>
        </div>
      );
    }
  };
  const handleClick = throttle(() => {
    onClick && onClick(itemData);
  }, 300);
  return (
    <div
      className={`${b()} ${isReceived ? disabdb() : ""} ${className} `}
      {...rest}
    >
      <div
        className={b("box", { small: type === "small" })}
        style={{
          width: couponWidth,
          height: couponHeight,
          backgroundImage: `url(${couponBgImg})`,
          marginRight: `${couponMarginRight}px`,
          marginBottom: `${couponMarginBottom}px`,
        }}
      >
        <div
          className={b("main")}
          style={{ width: couponMainWidth, color: couponMainColor }}
        >
          {renderPrice()}
          <div className={b("main-content")}>
            <p className={b("main-content--maintitle")}>{coupon?.mainTitle}</p>
            <p className={b("main-content--subtitle")}>{coupon?.subTitle}</p>
            {coupon?.timeRange && (
              <p className={b("main-content--timerange")}>
                {coupon?.timeRange}
              </p>
            )}
          </div>
          {isReceived && usedIcon && (
            <div className={b("main-usedicon")}>{usedIcon}</div>
          )}
        </div>
        <div className={b("btns")}>
          {type === "small" ? (
            <div className={b("btns-vertcal")} onClick={handleClick}>
              {btnText}
            </div>
          ) : (
            <Button
              {...buttonProps}
              disabled={isReceived}
              onClick={handleClick}
            >
              {btnText}
            </Button>
          )}
        </div>
        {coupon?.label && <div className={b("label")}>{coupon?.label}</div>}
      </div>
    </div>
  );
});

Coupon.defaultProps = defaultProps;
Coupon.displayName = "NutCoupon";
