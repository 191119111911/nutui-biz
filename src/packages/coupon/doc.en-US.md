# Coupon

### Intro

For coupon display

### Install

```javascript
import { Button } from "@nutui/nutui-react";
import { Coupon } from "@nutui/nutui-biz";
```

## Demo

### Basic Usage

:::demo

```ts
import React, { useState, CSSProperties } from "react";
import { ButtonProps } from "@nutui/nutui-react";
import { Coupon } from "@nutui/nutui-biz";

const App = () => {
  const buttonProps: Partial<ButtonProps> = React.useMemo(() => {
    return {
      type: "primary",
      size: "small",
      plain: true,
      className: "cancel-btn",
    };
  }, []);

  const usedIcon = React.useMemo(() => {
    return (
      <img
        src="https://storage.360buyimg.com/jdcdkh/open/1.0.0/assets/use-mask.60dc7c10.png"
        width="45px"
        height="42px"
      />
    );
  }, []);

  const baseCouponObj = React.useMemo(() => {
    return {
      price: "9",
      currency: "¥",
      mainTitle: "Available at 100 yuan or more",
      subTitle: "Only full discount coupons can be purchased for test",
      label: (
        <div style={{ color: "red" }}>Exclusive for internal purchase</div>
      ),
      timeRange: "2022.03.01-2022.04.01",
    };
  }, []);

  const couponBaseStyle: CSSProperties = React.useMemo(() => {
    return {
      width: "100%",
      height: "auto",
      backgroundImage: `url(https://storage.360buyimg.com/jdcdkh/open/1.0.0/assets/bg-coupon-red.f6ae2e19.png)`,
    };
  }, []);

  const couponMainBaseStyle: CSSProperties = React.useMemo(() => {
    return {
      width: "69%",
      color: "#fff",
    };
  }, []);

  const [btnText, setBtnText] = useState("Get it now");

  const [receivedStatus, setReceivedStatus] = useState(false);
  const basedOnClick = React.useCallback(() => {
    setBtnText("Received");
    setReceivedStatus(true);
  }, [btnText, receivedStatus]);
  const demoStyle = {
    height: "100%",
    overflow: "auto",
    padding: "17px 17px 0 17px",
  };
  return (
    <div style={demoStyle}>
      <Coupon
        pricePosition="back"
        couponStyle={couponBaseStyle}
        couponMainStyle={couponMainBaseStyle}
        couponData={baseCouponObj}
        btnText={btnText}
        isReceived={receivedStatus}
        usedIcon={usedIcon}
        buttonProps={buttonProps}
        onBtnClick={basedOnClick}
      ></Coupon>
    </div>
  );
};
export default App;
```

:::

### Coupon components of small card type

:::demo

```ts
import React, { CSSProperties, useState } from "react";
import { Coupon } from "@nutui/nutui-biz";

const App = () => {
  const couponSmallStyle: CSSProperties = React.useMemo(() => {
    return {
      width: "127px",
      height: "auto",
      backgroundImage: `url(https://static.360buyimg.com/jdcdkh/open/1.0.0/assets/bg-coupon.6df5b4ed.png)`,
      marginRight: `10px`,
      marginBottom: `10px`,
    };
  }, []);
  const couponMainSmallStyle: CSSProperties = React.useMemo(() => {
    return {
      width: "80%",
      color: "red",
    };
  }, []);
  const couponObj = React.useMemo(() => {
    return {
      price: 9,
      currency: "¥",
      mainTitle: "Available at 100 yuan or more",
      subTitle: "Only full discount coupons can be purchased for test",
      label: "618",
    };
  }, []);
  const usedIcon = React.useMemo(() => {
    return (
      <img
        src="https://storage.360buyimg.com/jdcdkh/open/1.0.0/assets/use-mask.60dc7c10.png"
        width="45px"
        height="42px"
      />
    );
  }, []);

  const [arrReceived, setArrReceived] = useState<Array<number>>([]);
  const receivedBtn = React.useCallback(
    (item) => {
      console.log(item);
      if (!arrReceived.includes(item.item)) {
        arrReceived.push(item.item);
      }
      setArrReceived([...arrReceived]);
    },
    [arrReceived]
  );

  return (
    <div
      style={{
        height: "100%",
        overflow: "auto",
        padding: "17px 17px 0 17px",
      }}
    >
      <div
        style={{
          display: "flex",
          flexDirection: "row",
          flexWrap: "wrap",
          width: "700px",
        }}
      >
        {[1, 2, 3, 4, 5, 6, 7, 8, 9, 10].map((item) => {
          return (
            <Coupon
              key={item}
              pricePosition="front"
              type="small"
              usedIcon={usedIcon}
              isReceived={arrReceived.includes(item)}
              couponMainStyle={couponMainSmallStyle}
              couponStyle={couponSmallStyle}
              couponData={{ ...couponObj, item }}
              btnText={arrReceived.includes(item) ? "Received" : "Get it now"}
              onBtnClick={receivedBtn}
            ></Coupon>
          );
        })}
      </div>
    </div>
  );
};
export default App;
```

:::

## API

### Props

| Attribute       | Description                                               | Type                 | Default     |
| --------------- | --------------------------------------------------------- | -------------------- | ----------- |
| type            | Type of coupon Optional values are `large` `small`        | String               | `large`     |
| couponStyle     | Style of each coupon                                      | CSSProperties        | --          |
| couponMainStyle | The style of the coupon body                              | CSSProperties        | --          |
| pricePosition   | Front and back position of price and label `front` `back` | String               | `back`      |
| couponData      | Render coupon content                                     | ICouponType          | -           |
| btnText         | content of Button                                         | Stringe              | -           |
| isReceived      | Whether to receive coupons                                | Boolean              | -           |
| receivedBtnText | content of after receiving                                | String               | `full-line` |
| buttonProps     | the props of buttton，from nutui-react                    | Partial<ButtonProps> | --          |
| usedIcon        | Received icon                                             | Boolean              | `true`      |

### ICouponType

| Attribute | Description                                          | Type           | Default |
| --------- | ---------------------------------------------------- | -------------- | ------- |
| price     | The price of the coupon                              | string \| number | -       |
| currency  | Currency symbol                                      | string         | -       |
| mainTitle | Main title                                           | string         | -       |
| subTitle  | Subtitle                                             | string         | -       |
| timeRange | Time range of coupon use                             | string         | -       |
| label     | Label content in the upper left corner of the coupon | ReactNode      | -       |

### Events

| Attribute | Description                                                                             | Arguments  |
| --------- | --------------------------------------------------------------------------------------- | ---------- |
| onClick   | Click Event，The data passed by the parent component is returned to the parent componen | couponData |
