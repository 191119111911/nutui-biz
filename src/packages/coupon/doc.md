#  Address 地址

### 介绍

地址选择

