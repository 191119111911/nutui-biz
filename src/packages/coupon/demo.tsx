import React, { useState, CSSProperties } from "react";
import { Coupon } from "./coupon";
import { useTranslate } from "../../sites/assets/locale";
import { ButtonProps } from "@nutui/nutui-react";
interface T {
  basic: string;
  mulText: string;
  btnText: string;
  receivedBtnText: string;
  mainTitle: string;
  subTitle: string;
  label: string;
  mulDesc: string;
}

const CouponDemo = () => {
  const [translated] = useTranslate<T>({
    "zh-CN": {
      basic: "基本用法",
      mulText: "小卡片类型的优惠组件",
      btnText: "立即领取",
      receivedBtnText: "已领取",
      mainTitle: "满100元可用",
      subTitle: "仅可购买满折券测试",
      label: "内购专享",
      mulDesc: "多行优惠券，在组件外层包裹元素上设置样式",
    },
    "en-US": {
      basic: "Basic Usage",
      mulText: "Coupon components of small card type",
      btnText: "Get it now",
      receivedBtnText: "Received",
      mainTitle: "Main title content",
      subTitle: "Used to render subtitle text",
      label: "Exclusive for internal purchase",
      mulDesc:
        "Multiline coupon, set the style on the outer package element of the component",
    },
  });

  //组件共有变量
  //按钮props
  const buttonProps: Partial<ButtonProps> = React.useMemo(() => {
    return {
      type: "primary",
      size: "small",
      plain: true,
      className: "cancel-btn",
    };
  }, []);

  //已经使用的icon标记
  const usedIcon = React.useMemo(() => {
    return (
      <img
        src="https://storage.360buyimg.com/jdcdkh/open/1.0.0/assets/use-mask.60dc7c10.png"
        width="45px"
        height="42px"
      />
    );
  }, []);
  //渲染组件文案内容
  const baseCouponObj = React.useMemo(() => {
    return {
      price: "9.1234123123789123712893",
      currency: "¥",
      mainTitle: translated.mainTitle,
      subTitle: translated.subTitle,
      label: translated.label,
      timeRange: "2022.03.01-2022.04.01",
    };
  }, []);

  //基本用法
  const couponBaseStyle: CSSProperties = React.useMemo(() => {
    return {
      width: "100%",
      height: "auto",
      backgroundImage: `url(https://storage.360buyimg.com/jdcdkh/open/1.0.0/assets/bg-coupon-red.f6ae2e19.png)`,
    };
  }, []);
  const couponMainBaseStyle: CSSProperties = React.useMemo(() => {
    return {
      width: "69%",
      color: "#fff",
    };
  }, []);
  //按钮文案
  const [btnText, setBtnText] = useState(translated.btnText);
  //是否点击了立即领取按钮
  const [receivedStatus, setReceivedStatus] = useState(false);

  const basedOnClick = React.useCallback(() => {
    setBtnText(translated.receivedBtnText);
    setReceivedStatus(true);
  }, [btnText, receivedStatus]);

  //多行展示小卡片优惠券组件
  const couponSmallStyle: CSSProperties = React.useMemo(() => {
    return {
      width: "127px",
      height: "auto",
      backgroundImage: `url(https://static.360buyimg.com/jdcdkh/open/1.0.0/assets/bg-coupon.6df5b4ed.png)`,
      marginRight: `10px`,
      marginBottom: `10px`,
    };
  }, []);
  const couponMainSmallStyle: CSSProperties = React.useMemo(() => {
    return {
      width: "80%",
      color: "red",
    };
  }, []);

  const couponObj = React.useMemo(() => {
    return {
      price: 9,
      currency: "¥",
      mainTitle: translated.mainTitle,
      subTitle: translated.subTitle,
      label: "618",
    };
  }, []);
  const receivedBtn = React.useCallback((item: number) => {
    console.log(item);
  }, []);
  return (
    <>
      <div className="demo">
        <h2>{translated.basic}</h2>
        <Coupon
          pricePosition="back"
          couponStyle={couponBaseStyle}
          couponMainStyle={couponMainBaseStyle}
          coupon={baseCouponObj}
          btnText={btnText}
          isReceived={receivedStatus}
          usedIcon={usedIcon}
          buttonProps={buttonProps}
          onBtnClick={basedOnClick}
        ></Coupon>
        <h2>{translated.mulText}</h2>
        <div style={{ width: "100%", overflow: "scroll" }}>
          <div
            style={{
              display: "flex",
              flexDirection: "row",
              flexWrap: "wrap",
              width: "700px",
            }}
          >
            {[1, 2, 3, 4, 5, 6, 7, 8, 9, 10].map((item) => {
              return (
                <Coupon
                  key={item}
                  type="small"
                  couponMainStyle={couponMainSmallStyle}
                  couponStyle={couponSmallStyle}
                  coupon={couponObj}
                  buttonProps={buttonProps}
                  itemData={item}
                  btnText={translated.btnText}
                  onBtnClick={receivedBtn}
                ></Coupon>
              );
            })}
          </div>
        </div>
      </div>
    </>
  );
};

export default CouponDemo;
