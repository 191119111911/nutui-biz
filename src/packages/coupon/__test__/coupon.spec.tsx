import * as React from "react";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Coupon, ButtonPropsType } from "../coupon";

test("should match snapshot", () => {
  const basedOnClick = jest.fn();
  const buttonProps: Partial<ButtonPropsType> = {
    type: "primary",
    size: "small",
    plain: true,
    className: "cancel-btn",
  };
  const state = {
    //优惠券样式
    couponBaseStyle: {
      width: "100%",
      height: "auto",
      backgroundImage: `url(https://storage.360buyimg.com/jdcdkh/open/1.0.0/assets/bg-coupon-red.f6ae2e19.png)`,
    },
    couponMainBaseStyle: {
      width: "69%",
      color: "#fff",
    },
    baseCouponObj: {
      price: "9",
      currency: "¥",
      mainTitle: "满100元可用",
      subTitle: "仅可购买满折券测试",
      label: "内购专享",
      timeRange: "2022.03.01-2022.04.01",
    },
    btnText: "立即领取",
    receivedStatus: false,
    usedIcon: (
      <img
        src="https://storage.360buyimg.com/jdcdkh/open/1.0.0/assets/use-mask.60dc7c10.png"
        width="45px"
        height="42px"
      />
    ),
    buttonProps: buttonProps,
  };
  const { asFragment } = render(
    <>
      <Coupon
        pricePosition="back"
        couponStyle={state.couponBaseStyle}
        couponMainStyle={state.couponMainBaseStyle}
        couponData={state.baseCouponObj}
        btnText={state.btnText}
        isReceived={state.receivedStatus}
        usedIcon={state.usedIcon}
        buttonProps={state.buttonProps}
        onBtnClick={basedOnClick}
      ></Coupon>
    </>
  );
  expect(asFragment()).toMatchSnapshot();
});

test("prop of Basic Usage", () => {
  const basedOnClick = jest.fn();
  const buttonProps: Partial<ButtonPropsType> = {
    type: "primary",
    size: "small",
    plain: true,
    className: "cancel-btn",
  };
  const state = {
    //优惠券样式
    couponBaseStyle: {
      width: "100%",
      height: "auto",
      backgroundImage: `url(https://storage.360buyimg.com/jdcdkh/open/1.0.0/assets/bg-coupon-red.f6ae2e19.png)`,
    },
    couponMainBaseStyle: {
      width: "69%",
      color: "#fff",
    },
    baseCouponObj: {
      price: "9",
      currency: "¥",
      mainTitle: "满100元可用",
      subTitle: "仅可购买满折券测试",
      label: "内购专享",
      timeRange: "2022.03.01-2022.04.01",
    },
    btnText: "立即领取",
    receivedStatus: false,
    usedIcon: (
      <img
        src="https://storage.360buyimg.com/jdcdkh/open/1.0.0/assets/use-mask.60dc7c10.png"
        width="45px"
        height="42px"
      />
    ),
    buttonProps: buttonProps,
  };
  const { container } = render(
    <>
      <Coupon
        pricePosition="front"
        couponStyle={state.couponBaseStyle}
        couponMainStyle={state.couponMainBaseStyle}
        couponData={state.baseCouponObj}
        btnText={state.btnText}
        isReceived={state.receivedStatus}
        usedIcon={state.usedIcon}
        buttonProps={state.buttonProps}
        onBtnClick={basedOnClick}
      ></Coupon>
    </>
  );
  //价格和单位的前后位置
  expect(
    container.querySelectorAll(".nb-coupon .nb-coupon__main-price div")[1]
      ?.innerHTML
  ).toBe("¥");
  //点击按钮事件
  const nutCoupon = container.querySelector(
    ".nb-coupon .nb-coupon__btns .nut-button--primary"
  ) as HTMLElement;
  fireEvent.click(nutCoupon);
  expect(basedOnClick).toBeCalled();
});

test("small type of coupon", () => {
  const basedOnClick = jest.fn();
  const buttonProps: Partial<ButtonPropsType> = {
    type: "primary",
    size: "small",
    plain: true,
    className: "cancel-btn",
  };
  const state = {
    //优惠券样式
    couponBaseStyle: {
      width: "100%",
      height: "auto",
      backgroundImage: `url(https://storage.360buyimg.com/jdcdkh/open/1.0.0/assets/bg-coupon-red.f6ae2e19.png)`,
    },
    couponMainBaseStyle: {
      width: "69%",
      color: "#fff",
    },
    baseCouponObj: {
      price: "9",
      currency: "¥",
      mainTitle: "满100元可用",
      subTitle: "仅可购买满折券测试",
      label: "内购专享",
      timeRange: "2022.03.01-2022.04.01",
    },
    btnText: "立即领取",
    receivedStatus: true,
    usedIcon: (
      <img
        src="https://storage.360buyimg.com/jdcdkh/open/1.0.0/assets/use-mask.60dc7c10.png"
        width="45px"
        height="42px"
      />
    ),
    buttonProps: buttonProps,
  };
  const { container } = render(
    <>
      <Coupon
        type="small"
        pricePosition="front"
        couponStyle={state.couponBaseStyle}
        couponMainStyle={state.couponMainBaseStyle}
        couponData={state.baseCouponObj}
        btnText={state.btnText}
        isReceived={state.receivedStatus}
        usedIcon={state.usedIcon}
        buttonProps={state.buttonProps}
        onBtnClick={basedOnClick}
      ></Coupon>
    </>
  );

  expect(
    container.querySelector(".nb-coupon .nb-coupon__btns div")
  ).toHaveClass("nb-coupon__btns-vertcal");
});

test("test price length", () => {
  const basedOnClick = jest.fn();
  const buttonProps: Partial<ButtonPropsType> = {
    type: "primary",
    size: "small",
    plain: true,
    className: "cancel-btn",
  };
  const state = {
    //优惠券样式
    couponBaseStyle: {
      width: "100%",
      height: "auto",
      backgroundImage: `url(https://storage.360buyimg.com/jdcdkh/open/1.0.0/assets/bg-coupon-red.f6ae2e19.png)`,
    },
    couponMainBaseStyle: {
      width: "69%",
      color: "#fff",
    },
    baseCouponObj: {
      price: "9.123",
      currency: "¥",
      mainTitle: "满100元可用",
      subTitle: "仅可购买满折券测试",
      label: "内购专享",
      timeRange: "2022.03.01-2022.04.01",
    },
    btnText: "立即领取",
    receivedStatus: false,
    usedIcon: (
      <img
        src="https://storage.360buyimg.com/jdcdkh/open/1.0.0/assets/use-mask.60dc7c10.png"
        width="45px"
        height="42px"
      />
    ),
    buttonProps: buttonProps,
  };
  const { container } = render(
    <>
      <Coupon
        pricePosition="front"
        couponStyle={state.couponBaseStyle}
        couponMainStyle={state.couponMainBaseStyle}
        couponData={state.baseCouponObj}
        btnText={state.btnText}
        isReceived={state.receivedStatus}
        usedIcon={state.usedIcon}
        buttonProps={state.buttonProps}
        onBtnClick={basedOnClick}
      ></Coupon>
    </>
  );
  //价格和单位的前后位置
  expect(
    container.querySelector(".nb-coupon .nb-coupon__main-price--number")
      ?.textContent
  ).toBe("9.12");
});

test("test price length sec", () => {
  const basedOnClick = jest.fn();
  const buttonProps: Partial<ButtonPropsType> = {
    type: "primary",
    size: "small",
    plain: true,
    className: "cancel-btn",
  };
  const state = {
    //优惠券样式
    couponBaseStyle: {
      width: "100%",
      height: "auto",
      backgroundImage: `url(https://storage.360buyimg.com/jdcdkh/open/1.0.0/assets/bg-coupon-red.f6ae2e19.png)`,
    },
    couponMainBaseStyle: {
      width: "69%",
      color: "#fff",
    },
    baseCouponObj: {
      price: "9.1",
      currency: "¥",
      mainTitle: "满100元可用",
      subTitle: "仅可购买满折券测试",
      label: "内购专享",
      timeRange: "2022.03.01-2022.04.01",
    },
    btnText: "立即领取",
    receivedStatus: false,
    usedIcon: (
      <img
        src="https://storage.360buyimg.com/jdcdkh/open/1.0.0/assets/use-mask.60dc7c10.png"
        width="45px"
        height="42px"
      />
    ),
    buttonProps: buttonProps,
  };
  const { container } = render(
    <>
      <Coupon
        pricePosition="front"
        couponStyle={state.couponBaseStyle}
        couponMainStyle={state.couponMainBaseStyle}
        couponData={state.baseCouponObj}
        btnText={state.btnText}
        isReceived={state.receivedStatus}
        usedIcon={state.usedIcon}
        buttonProps={state.buttonProps}
        onBtnClick={basedOnClick}
      ></Coupon>
    </>
  );
  //价格和单位的前后位置
  expect(
    container.querySelector(".nb-coupon .nb-coupon__main-price--number")
      ?.textContent
  ).toBe("9.1");
});
