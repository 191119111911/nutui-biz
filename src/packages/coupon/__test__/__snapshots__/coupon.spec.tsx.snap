// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="nut-biz-coupon  undefined "
  >
    <div
      class="nut-biz-coupon__box"
      style="width: 100%; height: auto; background-image: url(https://storage.360buyimg.com/jdcdkh/open/1.0.0/assets/bg-coupon-red.f6ae2e19.png);"
    >
      <div
        class="nut-biz-coupon__main"
        style="width: 69%; color: rgb(255, 255, 255);"
      >
        <div
          class="nut-biz-coupon__main-price"
        >
          <div
            class="nut-biz-coupon__main-price--currency"
          >
            ¥
          </div>
          <div
            class="nut-biz-coupon__main-price--number"
          >
            9.00
          </div>
        </div>
        <div
          class="nut-biz-coupon__main-content"
        >
          <p
            class="nut-biz-coupon__main-content--maintitle"
          >
            满100元可用
          </p>
          <p
            class="nut-biz-coupon__main-content--subtitle"
          >
            仅可购买满折券测试
          </p>
          <p
            class="nut-biz-coupon__main-content--timerange"
          >
            2022.03.01-2022.04.01
          </p>
        </div>
      </div>
      <div
        class="nut-biz-coupon__btns"
      >
        <button
          class="nut-button nut-button--primary nut-button--small nut-button--round nut-button--plain cancel-btn"
        >
          <div
            class="nut-button__warp"
          >
            <div
              class=""
            >
              立即领取
            </div>
          </div>
        </button>
      </div>
      <div
        class="nut-biz-coupon__label"
      >
        内购专享
      </div>
    </div>
  </div>
</DocumentFragment>
`;
